C7Engine/EntryPoints/CityInteractions.cs

```diff
@@ -21,5 +21,16 @@ public static void BuildCity(int x, int y, string playerGuid, string name)
 			tileWithNewCity.cityAtTile = newCity;
 			tileWithNewCity.overlays.road = true;
 		}
+
+		public static void DestroyCity(int x, int y) {
+			Tile tile = EngineStorage.gameData.map.tileAt(x, y);
+			tile.DisbandNonDefendingUnits();
+			while (tile.cityAtTile.size > 0) {
+				tile.cityAtTile.RemoveCitizen();
```

Comment:
seems like we could use a convenience function RemoveCitizens(int n) and maybe RemoveAllCitizens()

Reply:
Added both of them, and it does indeed clean it up a bit.  It also provides a bounds check in the RemoveCitizen(int n) method which protects against problems caused by removing too many citizens, so that is nice.